onnx/defs/map/defs.cc

```diff
@@ -0,0 +1,402 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "onnx/defs/function.h"
+#include "onnx/defs/schema.h"
+
+#include <algorithm>
+#include <numeric>
+
+namespace ONNX_NAMESPACE {
+static std::vector<std::string> all_map_types() {
+    auto map_key_types = OpSchema::all_map_key_types();
+    auto map_value_types = OpSchema::all_tensor_sequence_types();
+    static std::vector<std::string> all_map_types;
+    for (auto key_type : map_key_types) {
+        for (auto value_type : map_value_types) {
+            std::string map_type = "map("+ key_type + ", " + value_type + ")";
+            all_map_types.emplace_back(map_type);
+        }
+    }
+    return all_map_types;
+}
+
+static const char* MapConstruct_ver18_doc = R"DOC(
+Constructs either an empty map of a certain type specified by the key_type and value_type attributes,
+or a map structure with specified 'keys' and 'values'
+'keys' and 'values' must have the same number of elements.
+All 'keys' must have the same data type.
+All 'values' must all be of the same type (tensor, sequence or map) and have the same data type.
```

Comment:
It's a bit unclear what's supported as values (seeing description below as well).